### ISSUE TITLE
Unignorable "invalid-name" when accessing "private" members

### ISSUE THREAD

**Issue by wams3** [pylint-dev/pylint#8885]
### Bug description

Given are two files main.py and server.py in a module "mod".

test.py contains the unit tests for server.py and accesses a "private" member of the class Server.
This access leads to "invalid-name" but can not be ignored by adding disable statements:

test.py:
```
#pylint: disable=protected-access, missing-module-docstring, missing-function-docstring, missing-class-docstring, too-few-public-methods
import mod.server

class TestServer():

    def test_server(self):
        # pylint: disable=invalid-name
        server = mod.server.Server()
        server._Server__dlt_socket = 1 # pylint: disable=invalid-name

```

server.py:
```
#pylint: disable=missing-module-docstring, missing-function-docstring, missing-class-docstring, too-few-public-methods, unused-private-member

class Server():
    def __init__(self):
        self.__dlt_socket = False

```

### Command used

```shell
pylint mod/
```

### Pylint output

```shell
************* Module mod.test
mod\test.py:9:8: C0103: Attribute name "_Server__dlt_socket" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

Expected behavior is to at least be able to ignore the message.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.6
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
```
